lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -554,6 +557,18 @@ newDBLayer trace fp timeInterpreter = do
                 let cpt = CertificatePublicationTime {slotNo, slotInternalIndex}
                 pure (cpt, cert)
 
+        putHeader point =
+                let record = mkBlockHeader point
+                    key = TH.blockHeight record
+                in repsert (BlockHeaderKey key) record
+
+        listHeaders k = do
+            reverse . fmap (fromBlockHeaders . entityVal) <$> selectList [ ]
+                [ Desc BlockHeight
+                , LimitTo k
+                ]
+
+        dropHeaders = deleteWhere @_ @_ @TH.BlockHeader []
```

Comment:
That sounds a little drastic no :) ? It seems like there's a conflict regarding the goal of such function. To me, the dropHeaders / prune function should remove block headers that are no-longer needed (i.e., those that are older than `k`). 

What is the rationale of the current implementation above?

Reply:
I currently have no plans implementing prune. I think the size of the header table is fairly ok, is it not? `dropHeaders` was used previously in the implementation where I kept only one block header in total in the table. This is not the case anymore, so I could remove it altogether.